internal/httpapi/httpapi.go

```diff
@@ -220,6 +222,30 @@ func (s *Server) handleReady(w http.ResponseWriter, r *http.Request, ps httprout
 	w.WriteHeader(http.StatusOK)
 }
 
+func (s *Server) handleEntity(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	re, err := s.reporter.ReportEntity()
+	if err != nil {
+		s.logger.WithError(err).Error("cannot report entity status")
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	b, jerr := json.Marshal(re)
+	if jerr != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		s.logger.WithError(jerr).Warn("couldn't encode entity report")
+		return
+	}
+
+	_, err = w.Write(b)
+	if err != nil {
+		s.logger.Warn("cannot write entity response, error: " + err.Error())
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
```

Comment:
shouldn't it be a case where id is not present yet and `204` is returned?

Reply:
I have the same question.

Reply:
last commit